cep18/src/entry_points.rs

```diff
@@ -0,0 +1,221 @@
+    EntryPoint::new(
+        String::from(ALLOWANCE_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(OWNER, Key::cl_type()),
+            Parameter::new(SPENDER, Key::cl_type()),
+        ],
+        U256::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `approve` entry point.
+pub fn approve() -> EntryPoint {
+    EntryPoint::new(
+        String::from(APPROVE_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(SPENDER, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `increase_allowance` entry point.
+pub fn increase_allowance() -> EntryPoint {
+    EntryPoint::new(
+        String::from(INCREASE_ALLOWANCE_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(SPENDER, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `decrease_allowance` entry point.
+pub fn decrease_allowance() -> EntryPoint {
+    EntryPoint::new(
+        String::from(DECREASE_ALLOWANCE_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(SPENDER, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `transfer` entry point.
+pub fn transfer() -> EntryPoint {
+    EntryPoint::new(
+        String::from(TRANSFER_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(RECIPIENT, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `balance_of` entry point.
+pub fn balance_of() -> EntryPoint {
+    EntryPoint::new(
+        String::from(BALANCE_OF_ENTRY_POINT_NAME),
+        vec![Parameter::new(ADDRESS, Key::cl_type())],
+        U256::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `total_supply` entry point.
+pub fn total_supply() -> EntryPoint {
+    EntryPoint::new(
+        String::from(TOTAL_SUPPLY_ENTRY_POINT_NAME),
+        Vec::new(),
+        U256::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `decimals` entry point.
+pub fn decimals() -> EntryPoint {
+    EntryPoint::new(
+        String::from(DECIMALS_ENTRY_POINT_NAME),
+        Vec::new(),
+        u8::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `burn` entry point.
+pub fn burn() -> EntryPoint {
+    EntryPoint::new(
+        String::from(BURN_ENTRY_POINT_NAME),
+        vec![Parameter::new(AMOUNT, U256::cl_type())],
```

Comment:
```suggestion
        vec![
          Parameter::new(OWNER, Key::cl_type()),
          Parameter::new(AMOUNT, U256::cl_type()),
        ],
```

owner should be added